### PR TITLE
Follow the user’s tab-size setting where possible

### LINKED ISSUE #4832: It overrides GitHub built-in "Tab size preference"
### Please ensure:

- [X] The bug is caused by Refined GitHub. It doesn't happen if I disable the extension.

### Description

If enabled this extension, it will override GitHub built-in "Tab size preference".

### Screenshot

I use 3 spaces as tab size when rendering tabs and it's set in "Tab size preference" of GitHub "Appearance" settings.

When disabled this extension:

图片

When enabled this extension:

图片

### Console errors

_No response_

### Example URL

https://github.com/sindresorhus/refined-github/blob/main/.xo-config.json

### Browser(s) used

Firefox

### PR DESCRIPTION
**Pull request by fregante**
- Fixes #4832 
- Fixes https://github.com/sindresorhus/refined-github/issues/4806

## What

- In files, follow the user’s setting
- In comments, stick to 4 characters because GitHub is ignoring the setting
- Preserves support in `show-whitespace`

### Working natively

- https://github.com/sindresorhus/refined-github/blob/a52decfe2d0b5cd4617c08055bcf4a0a23812e4e/package.json
- https://github.com/sindresorhus/refined-github/pull/4833/files

### GitHub bugs…

GitHub ignores the user setting in comments and just sets `data-tab-size="8"` in the DOM 🤷‍♂️  so I will force it in this case.

- Review: https://github.com/sindresorhus/refined-github/pull/4833#discussion_r719986599

- Inlined code:

	https://github.com/sindresorhus/refined-github/blob/a52decfe2d0b5cd4617c08055bcf4a0a23812e4e/package.json#L1-L6

- Code block
	  
	```js
	{
		no: "support"
	}
	```
- Try editing a comment and press <kbd>tab</tab>

# Screenshot 1

> Screen Shot 2

# Screenshot 2
 
> Screen Shot 3

### CODE REVIEW

**Comment by kidonng** [comment]
Also fixes #4806

File: source/refined-github.css [hunk 1]
```diff
@@ -5,12 +5,6 @@
 	--github-border-color: var(--color-border-primary, #e1e4e8);
 }
 
-/* Make tab indented code more readable on GitHub and Gist */
```
Comment (busches) [inline source/refined-github.css:8]: Can we scope it just to enterprise for now, so all of us don't have to add in custom css?

Comment (fregante) [reply source/refined-github.css:8]: Sure, is it possible?

File: source/features/show-whitespace.css [hunk 2]
```diff
@@ -1,3 +1,33 @@
+/*
+Can't use this because attributes are strings and calc() won't cast them
+ttps://stackoverflow.com/a/40815884
+
+[data-tab-size] {
+	--tab-size: attr(data-tab-size);
+}
```
Comment (fregante) [inline source/features/show-whitespace.css:7]: Demo suggestion, do not resolve

```suggestion
[data-tab-size] {

	--tab-size: attr(data-tab-size);
}
```

File: source/refined-github.css (hunk 1 above)
Comment (fregante) [reply source/refined-github.css:8]: Done

**Comment by fregante** [comment]
Last chance for review. Will merge within 8 hours

File: source/features/show-whitespace.css [hunk 3]
```diff
@@ -1,3 +1,32 @@
+/*
+Can't use this because attributes are strings and calc() won't cast them
+ttps://stackoverflow.com/a/40815884
```
Comment (yakov116) [inline source/features/show-whitespace.css:3]: Should this file get a GHE comment?

**Comment by fregante** [comment]
argh I forgot to merge this… I'll release v21.8.2 I guess 😂

File: source/features/show-whitespace.css (hunk 3 above)
Comment (fregante) [reply source/features/show-whitespace.css:3]: GHE is mentioned in a different comment. This code is not related to GHE

Comment (kidonng) [reply source/features/show-whitespace.css:3]: Isn't this link missing a leading `h`?

**Pull request merged**